docs/database_migration.md

```diff
@@ -6,43 +6,53 @@ NOTE: [pgloader](https://github.com/dimitri/pgloader) is a tool that can easily
 
 The process is as follows:
 
-1. Create a dump of existing data.
-2. Change all `CREATE TABLE ...` lines to `DELETE FROM tablename;`. We will later have LLDAP create the schema for us, so we want to clear out existing data to replace it with the original data.
-3. Do any syntax fixes for the target db syntax
-4. Change your LLDAP config database_url to point to the new target and restart.
-5. After LLDAP has started, stop it.
-6. Execute the manicured dump file against the new database.
+1. Create empty schema on target database
+2. Dump existing values
+3. Sanitize for target DB (not always required)
+4. Insert data into target
+5. Change LLDAP config to new target
 
 The steps below assume you already have PostgreSQL or MySQL set up with an empty database for LLDAP to use.
 
-## Create a dump
+## Create schema on target
 
-First, we must dump the existing data to a file. The dump must be tweaked slightly according to your target db. See below for commands
+LLDAP has a command that will connect to a target database and initialize the
+schema. If running with docker, run the following command to use your active
+instance (this has the benefit of ensuring your container has access):
 
-### PostgreSQL
+```
+docker exec -it <LLDAP container name> /app/lldap create_schema -d <Target database url>
+```
+
+If it succeeds, you can proceed to the next step.
 
-PostgreSQL uses a different hex string format and doesn't support `PRAGMA`.
+## Create a dump of existing data
+
+We want to dump all existing values to some file. The dump should consist just INSERT
+statements. There are various ways to do this, but a simple enough way is filtering a
+whole database dump. For example:
 
 ```
-sqlite3 /path/to/lldap/config/users.db .dump | \
-sed -r -e "s/X'([[:xdigit:]]+'[^'])/'\\\x\\1/g" \
--e 's/^CREATE TABLE IF NOT EXISTS "([^"]*)".*/DELETE FROM \1;/' \
--e '/^PRAGMA.*/d' > /path/to/dump.sql
+sqlite3 /path/to/lldap/config/users.db .dump | grep "^INSERT" > /path/to/dump.sql
 ```
 
-### MySQL
```

Comment:
Maybe let's keep a not about MySQL to tell users that we didn't forget about them, but there's nothing to do.